pkg/network/encoding/format.go

```diff
@@ -301,6 +308,39 @@ func formatDNSStatsByDomain(stats map[string]map[network.QueryType]network.DNSSt
 	}
 	return m
 }
+func formatDNSStatsByDomain(stats map[string]map[network.QueryType]network.DNSStats, domainSet map[string]int) map[int32]*model.DNSStats {
+	m := make(map[int32]*model.DNSStats)
+	for d, bytype := range stats {
+
+		for _, stat := range bytype {
+
+			pos, ok := domainSet[d]
+			if !ok {
+				pos = len(domainSet)
+				domainSet[d] = pos
+			}
+			if ms, ok := m[int32(pos)]; ok {
+				for rcode, count := range stat.DNSCountByRcode {
+					ms.DnsCountByRcode[rcode] += count
+				}
+				ms.DnsFailureLatencySum += stat.DNSFailureLatencySum
+				ms.DnsSuccessLatencySum += stat.DNSSuccessLatencySum
+				ms.DnsTimeouts += stat.DNSTimeouts
+
+			} else {
+				var ms model.DNSStats
+				ms.DnsCountByRcode = stat.DNSCountByRcode
+				ms.DnsFailureLatencySum = stat.DNSFailureLatencySum
+				ms.DnsSuccessLatencySum = stat.DNSSuccessLatencySum
+				ms.DnsTimeouts = stat.DNSTimeouts
+
+				m[int32(pos)] = &ms
+			}
+			break
```

Comment:
Why are we doing a break here and taking only the first type if there are multiple types?

Reply:
Good catch. Yeah, this needs to be removed @derekwbrown